### PR TITLE
#1097-- DropDown: Adds require to DropDown when required, and adds er…

### PR DESCRIPTION
**Pull request by aleksandrjPersonal**
…ror message property/state

#### Pull request checklist

- [x ] Addresses an existing issue: #1097
- [ ] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

DropDown: Adds require to DropDown when required, and adds error message property/state

(give an overview)

DropDown:

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.Props.ts [hunk 1]
```diff
@@ -96,6 +96,10 @@ export interface IDropdownProps extends React.Props<Dropdown> {
    */
   isDisabled?: boolean;
 
+  /**
+   * Descriptive label for the Dropdown Error Message
+   */
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.Props.ts]: change fle.

**Comment by aleksandrjPersonal** [comment]
@dzearing Change file added

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -144,6 +147,14 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
         { isOpen && (
           onRenderContainer(this.props, this._onRenderContainer)
         ) }
+        {
+          errorMessage &&
+          <MessageBar
+            messageBarType={ MessageBarType.error }
+            onDismiss={ () => { console.log('test'); } }>
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: probably want to remove this console.log :)

**Comment by mikewheaton** [comment]
Do we have designs for this? I'm surprised to see MessageBar used, when we have a simple piece of text for all other form controls' error states.

**Comment by micahgodbolt** [comment]
@mikewheaton hah, you can blame me for suggesting messageBar. I was thinking "why create custom text when we already have a message bar". If there is a design precedent elsewhere, I'm happy to suggest using that.

**Comment by mikewheaton** [comment]
I don't think it's necessarily wrong. We'd just want to make sure it looks okay in the context of a form with other error states, and run it past a designer.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -144,6 +147,14 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
         { isOpen && (
           onRenderContainer(this.props, this._onRenderContainer)
         ) }
+        {
+          errorMessage &&
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: why would we be using a MessageBar in a DropDown? Screenshots?

**Comment by aleksandrjPersonal** [comment]
@betrue-final-final Please review this DropDown with error state
dropdownwitherror

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (aleksandrjPersonal) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: removing Message Bar

**Comment by aleksandrjPersonal** [comment]
error
Per ben

**Comment by aleksandrjPersonal** [comment]
required
required and error

**Comment by aleksandrjPersonal** [comment]
@betrue-final-final It appears the Label has a padding of 5px
I created a new bug to track this issue:
https://github.com/OfficeDev/office-ui-fabric-react/issues/1562

I will address that in a upcoming PR

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.scss [hunk 4]
```diff
@@ -110,6 +110,9 @@ $DropDown-item-height: 36px;
       border-color: $ms-color-contrastWhiteDisabled;
       color: $ms-color-contrastWhiteDisabled;
     }
+    &.error {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.scss]: State/modifier classes like this should be in the format `elementIsState`. So in this case `titleIsError`

Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.scss]: Also, is it necessary here? I see the same border color being applied down around line 150.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.scss [hunk 5]
```diff
@@ -149,6 +152,9 @@ $DropDown-item-height: 36px;
   overflow: hidden;
   white-space: nowrap;
   text-overflow: ellipsis;
+  &.error {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.scss]: Class should be `titleIsError`

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 6]
```diff
@@ -108,6 +109,9 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
       disabled = isDisabled;
     }
 
+    required = true;
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Are these leftover from testing? `required` and `errorMessage` should be coming from the props, not set inside here.

File: common/changes/1097_Dropown_Require_2017-04-13-18-31.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (mikewheaton) [inline common/changes/1097_Dropown_Require_2017-04-13-18-31.json]: Please remove this duplicate change file.

Comment (aleksandrjPersonal) [inline common/changes/1097_Dropown_Require_2017-04-13-18-31.json]: removed

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.scss (hunk 4 above)
Comment (aleksandrjPersonal) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.scss]: Yea, it's needed in both places

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.scss (hunk 5 above)
Comment (aleksandrjPersonal) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.scss]: fixed

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx (hunk 6 above)
Comment (aleksandrjPersonal) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: you are correct, removing

**Comment by micahgodbolt** [comment]
Looks like we've got some dropdown tests failing.

**Comment by ImenTurki** [comment]
can you share the implementation of this require to DropDown when required , in my version I dont have it .

**Comment by gioce90** [comment]
I'm using dropdown and combobox in a sharepoint's webpart, but with the version 6.143.0 (the default one for Sharepoint) the required attribute doesn't work. 
 also tried upgrading to the version 6.164.1, but still doesn't work.
Which version I have to use?

**Pull request merged**